mmdet3d/models/layers/torchsparse_block.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Sequence, Union
+
+from mmengine.model import BaseModule
+from torch import nn
+
+from mmdet3d.utils import OptConfigType
+from .torchsparse import IS_TORCHSPARSE_AVAILABLE
+
+if IS_TORCHSPARSE_AVAILABLE:
+    import torchsparse.nn as spnn
+    from torchsparse.tensor import SparseTensor
+else:
+    SparseTensor = None
+
+
+class TorchsparseConvModule(BaseModule):
+    """A torchsparse conv block that bundles conv/norm/activation layers.
+
+    Args:
+        in_channels (int): In channels of block.
+        out_channels (int): Out channels of block.
+        kernel_size (int or Tuple[int]): Kernel_size of block. Defaults to 3.
+        stride (int or Tuple[int]): Stride of the first block. Defaults to 1.
+        dilation (int): Dilation of block. Defaults to 1.
+        transposed (int):
+        init_cfg (:obj:`ConfigDict` or dict, optional): Initialization config.
+            Defaults to None.
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: Union[int, Sequence[int]],
+        stride: Union[int, Sequence[int]] = 1,
+        dilation: int = 1,
+        bias: bool = False,
+        transposed: bool = False,
+        init_cfg: OptConfigType = None,
+    ) -> None:
+        super().__init__(init_cfg)
+        self.net = nn.Sequential(
```

Comment:
Please check the initilization of these module. Is it compatible with BaseModule?

Reply:
The input of Last Relu is from Identity and residual. Can‘t simply use two BaseModule.